include/st_pipeline_api.h

```diff
@@ -189,10 +189,28 @@ enum st_frame_fmt {
  */
 #define ST_FRAME_FLAG_EXT_BUF (ST_BIT32(0))
 
+#define ST_MAX_PLANES (4)
+
+/** The structure info for external frame */
+struct st_ext_frame {
+  /** Virtual address of external frame */
+  void* addr[ST_MAX_PLANES];
+  /** IOVA of external frame (only for non-convert mode) */
+  st_iova_t iova[ST_MAX_PLANES];
+  /** Linesize of external frame */
+  uint32_t linesize[ST_MAX_PLANES];
+  /** Size of external frame */
+  size_t size;
+  /** Private data for user */
+  void* opaque;
+};
+
 /** The structure info for frame meta. */
 struct st_frame {
   /** frame buffer address */
-  void* addr;
+  void* addr[ST_MAX_PLANES];
+  /** frame buffer linesize */
+  uint32_t linesize[ST_MAX_PLANES];
```

Comment:
Add a behavior note if linesize is zero or linesize is invalid value(ex < width). Maybe we can add a "int st_frame_sanity_check(struct st_frame*)" helper to verify if the parameters in one st_frame is valid or not.